src/script_opt/ZAM/README.md

```diff
@@ -75,13 +75,12 @@ of the expression are modified by compiled scripts.
 
 ### Incompatibilities:
 
+* ZAM ignores `assert` statements.
+
```

Comment:
We had chatted via DM a bit, but assuming `assert` finds its way into `testing/btest` over time, do we need to mark every test that's using it (which might be most  simple ones over time) with a `@TEST-REQUIRES: test "${ZEEK_ZAM}" != "1"` ?

I suspect tests would still run even in ZAM mode, but nothing actually asserted/validated and potentially missing out on errors specific to running in ZAM mode.

Reply:
You have a good point.  What comes to mind is if indeed we start using it for routine tests, down the road adding a mechanism that tells script optimization "please include the asserts rather than discarding them", which we'd turn on for running on the test suite.  That way we get both the performance benefits and the testing benefits.